sources/packages/backend/apps/api/src/services/student/student.service.ts

```diff
@@ -494,6 +494,78 @@ export class StudentService extends RecordDataModelService<Student> {
     return searchQuery.getMany();
   }
 
+  /**
+   * Search students based on the search criteria.
+   * @param searchCriteria options to search by firstName,
+   * lastName, appNumber or sin.
+   * @returns list of students.
+   */
+  async searchStudentApplicationForInstitution(
```

Comment:
Cant we reuse the existing query searchStudentApplication to use the instituion id as an optional parameter and return?

Reply:
I don't like the idea of doing that because it would make a simple thing like a query to a more complex method with conditions that will be harder to maintain. Thoughts @ann-aot @dheepak-aot @andrewsignori-aot  @sh16011993 ?

Reply:
Maybe I am wrong and I did not tested it, but would the change be something about moving the inner/left and one where inside an if like below?
```ts
    if (institutionId) {
      searchQuery
        .innerJoin("application.location", "pirLocation")
        .innerJoin("pirLocation.institution", "pirInstitution")
        .andWhere(
          "(offeringInstitution.id = :institutionId or pirInstitution.id = :institutionId)",
          { institutionId },
        );
    }
```

Reply:
Ok. Changed.

Reply:
I noticed the ` if (institutionId)` twice in the method. Was it because the proposed code on the comment didn't work or because it looks cleaner?